cgen error for ` _ = none`

```bash
v run a.v
```
```
==================
/tmp/v_2625/a.01HMWV3D4QDHES1C1B00P00B2E.tmp.c:12858: error: '{' expected (got ";")
...
==================
(Use `v -cg` to print the entire error message)                    
builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```
```bash
cat a.v
```
```v
fn x() ?bool {
        return none
}                                                                  
_ = x() or {
        _ = none
        true
}
```
I have no idea what's happening here, I was experimenting as I'm still learning the language and then it asked me to report.

and I want to leverage this opportunity to ask an off-topic little doubt,
in structs, what `__global:` allows that `pub mut:` doesn't?
I tried them in various forms and both worked fine,
the docs talks about "parent module only" stuff with `pub mut:` but I don't get it they both work no matter what I try.
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/assign.v
+++ b/vlib/v/checker/assign.v
@@ -283,6 +283,9 @@ fn (mut c Checker) assign_stmt(mut node ast.AssignStmt) {
 							right.right.pos)
 					}
 				} else if left.kind == .blank_ident {
+					if right is ast.None {
+						c.error('cannot assign none to blank `_` identifier', left.pos)
+					}
 					left_type = right_type
 					node.left_types[i] = right_type
 					if node.op !in [.assign, .decl_assign] {

--- a/vlib/v/checker/assign.v
+++ b/vlib/v/checker/assign.v
@@ -283,8 +283,8 @@ fn (mut c Checker) assign_stmt(mut node ast.AssignStmt) {
 							right.right.pos)
 					}
 				} else if left.kind == .blank_ident {
-					if right is ast.None {
-						c.error('cannot assign none to blank `_` identifier', left.pos)
+					if !is_decl && mut right is ast.None {
+						c.error('cannot assign a `none` value to blank `_` identifier', right.pos)
 					}
 					left_type = right_type
 					node.left_types[i] = right_type

--- a/vlib/v/checker/assign.v
+++ b/vlib/v/checker/assign.v
@@ -284,7 +284,8 @@ fn (mut c Checker) assign_stmt(mut node ast.AssignStmt) {
 					}
 				} else if left.kind == .blank_ident {
 					if !is_decl && mut right is ast.None {
-						c.error('cannot assign a `none` value to blank `_` identifier', right.pos)
+						c.error('cannot assign a `none` value to blank `_` identifier',
+							right.pos)
 					}
 					left_type = right_type
 					node.left_types[i] = right_type

--- a/vlib/v/checker/tests/assign_blank_none_err.out
+++ b/vlib/v/checker/tests/assign_blank_none_err.out
@@ -1,5 +1,12 @@
-vlib/v/checker/tests/assign_blank_none_err.vv:2:2: error: cannot assign none to blank `_` identifier
+vlib/v/checker/tests/assign_blank_none_err.vv:2:7: error: cannot assign a `none` value to a variable
     1 | fn main() {
-    2 |     _ = none
-      |     ^
-    3 | }
+    2 |     _ := none
+      |          ~~~~
+    3 |     _ = none
+    4 | }
+vlib/v/checker/tests/assign_blank_none_err.vv:3:6: error: cannot assign a `none` value to blank `_` identifier
+    1 | fn main() {
+    2 |     _ := none
+    3 |     _ = none
+      |         ~~~~
+    4 | }

--- a/vlib/v/checker/tests/assign_blank_none_err.out
+++ b/vlib/v/checker/tests/assign_blank_none_err.out
@@ -1,0 +1,5 @@
+vlib/v/checker/tests/assign_blank_none_err.vv:2:2: error: cannot assign none to blank `_` identifier
+    1 | fn main() {
+    2 |     _ = none
+      |     ^
+    3 | }

--- a/vlib/v/checker/tests/assign_blank_none_err.vv
+++ b/vlib/v/checker/tests/assign_blank_none_err.vv
@@ -1,0 +1,3 @@
+fn main() {
+	_ = none
+}

--- a/vlib/v/checker/tests/assign_blank_none_err.vv
+++ b/vlib/v/checker/tests/assign_blank_none_err.vv
@@ -1,3 +1,4 @@
 fn main() {
+	_ := none
 	_ = none
 }

